yt_dlp/extractor/nekohacker.py

```diff
@@ -0,0 +1,125 @@
+from datetime import datetime
+import re
+
+from yt_dlp.utils import ExtractorError, determine_ext, extract_attributes, get_element_by_class, unified_strdate
+from .common import InfoExtractor
+
+
+class NekoHackerIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?nekohacker\.com/(?P<id>[a-zA-Z0-9-_]+)'
```

Comment:
Use `[\w-]+`, and we can exclude the path we know definitely isn't an album page
```suggestion
    _VALID_URL = r'https?://(?:www\.)?nekohacker\.com/(?P<id>(?!free-dl)[\w-]+)'
```

Reply:
paths that are not a album are handled via `if webpage is None:`, but sure

Reply:
This is a good point; maybe that `ExtractorError` should not be expected then, especially if we also check for embed pages, as mentioned in https://github.com/yt-dlp/yt-dlp/pull/7003#issuecomment-1537416654 https://github.com/yt-dlp/yt-dlp/pull/7003#issuecomment-1537467522